Sprint №12 HomeWork done.

Vozzhd opened the pull request:
Домашняя работа №1 по изменению темы приложения - дополнится после вебинара, на данный момент всё работает корректно. Из дополнений хочется добавить, чтобы при первом запуске приложения в переменную записывалось значение соответствующее текущей теме устройства (dark-true\light-false). На данный момент приложение изначально со светлой темой.

Vozzhd committed:
Sprint №12
Part 1:
Added:
-App
-darkMode switch realisation
Changed:
-darkMode switch

Vozzhd committed:
Sprint №12
Part 2:
Added:
-Button
-TitleText
Changed:
-searchScreen

Vozzhd committed:
Sprint №12 HomeWork done, need refactoring

Vozzhd committed:
Sprint №12 HomeWork done, need refactoring
Всё окончательно в работоспособном состоянии.
НА очереди - улучшение кода.

Vozzhd committed:
Sprint №12 HomeWork done
Added:
-Utulities
-Adapter
-HistorySearch
-Reuse RecyclerView
-ExtendFunction for RecyclerView
-Button "back" in activity_search.xml
Changed:
-Adapters
-ViewLogic
-Layouts
-Hierarchy

DmitrijEgorow commented:
Кнопка «Очистить историю» не всегда влезает


https://github.com/Vozzhd/PlaylistMaker/assets/36837067/c6bf671e-657f-46f0-9f42-4c0dc30d8883

DmitrijEgorow commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -42,14 +52,43 @@
+            android:id="@+id/clearHistoryButton"
+            style="@style/SmallButton"
+            android:layout_marginTop="24dp"
+            android:text="Очистить историю"
```
Лучше использовать string-res, иначе возникнут сложности при локализации приложения

Vozzhd replied:
Упустил из виду, хардкод был во время разработки, заменил)

DmitrijEgorow replied:
Можно использовать namespace tools (`tools:text=""`), чтобы смотреть только в preview в студии

Vozzhd commented:
> Кнопка «Очистить историю» не всегда влезает
> 
>  Screen.Recording.2024-02-26.at.12.16.57.mov

Исправил добавлением веса к RecyclerView, слепил эмулятор с мелким экраном, проверил - в работе.

Vozzhd committed:
Sprint №12 HomeWork after review
Changed:
-throwed text of "clearHistoryButton" to resources
-added weight to RecyclerView

Vozzhd committed:
Sprint №12 HomeWork after review
Add:
-string-res
-marginBottom for clearHistoryButton
